packages/shell-api/src/shard.spec.ts

```diff
@@ -2629,4 +2677,90 @@ describe('Shard', function () {
       });
     });
   });
+
+  describe('integration chunks', function () {
+    let serviceProvider: CliServiceProvider;
+    let instanceState: ShellInstanceState;
+    let sh: Shard;
+    const dbName = 'test';
+    const ns = `${dbName}.coll`;
+    const shardId = 'rs-shard1';
+
+    const [mongos, rs0, rs1] = startTestCluster(
+      'shard',
+      // shards: 0 creates a setup without any initial shards
+      { topology: 'sharded', shards: 0 },
+      {
+        topology: 'replset',
+        args: ['--replSet', `${shardId}-0`, '--shardsvr'],
+      },
+      { topology: 'replset', args: ['--replSet', `${shardId}-1`, '--shardsvr'] }
+    );
+
+    before(async function () {
+      serviceProvider = await CliServiceProvider.connect(
+        await mongos.connectionString(),
+        dummyOptions,
+        {},
+        new EventEmitter()
+      );
+      instanceState = new ShellInstanceState(serviceProvider);
+      sh = new Shard(instanceState.currentDb);
+
+      // check replset uninitialized
+      let members = await (
+        await sh._database.getSiblingDB('config').getCollection('shards').find()
+      )
+        .sort({ _id: 1 })
+        .toArray();
+      expect(members.length).to.equal(0);
+
+      // add new shards
+      expect(
+        (await sh.addShard(`${shardId}-0/${await rs0.hostport()}`)).shardAdded
+      ).to.equal(`${shardId}-0`);
+      expect(
+        (await sh.addShard(`${shardId}-1/${await rs1.hostport()}`)).shardAdded
+      ).to.equal(`${shardId}-1`);
+      members = await (
+        await sh._database.getSiblingDB('config').getCollection('shards').find()
+      )
+        .sort({ _id: 1 })
+        .toArray();
+      expect(members.length).to.equal(2);
+      await sh._database.getSiblingDB(dbName).dropDatabase();
+      await sh._database.getSiblingDB(dbName).createCollection('unsharded');
+      await sh.enableSharding(dbName);
+      await sh.shardCollection(ns, { key: 1 });
+    });
+
+    after(function () {
+      return serviceProvider.close(true);
+    });
+
+    it('shows a full chunk list when there are 20 or less chunks', async function () {
+      for (let i = 0; i < 19; i++) {
+        await sh.splitAt(ns, { key: i + 1 });
+      }
+      const chunks = (await sh.status()).value.databases.find(
+        (d) => d.database._id === 'test'
+      ).collections[ns].chunks;
+      expect(chunks.length).to.equal(20);
+    });
+
+    it('cuts a chunk list when there are more than 20 chunks', async function () {
+      await sh.splitAt(ns, { key: 20 });
+      const chunks = (await sh.status()).value.databases.find(
+        (d) => d.database._id === 'test'
+      ).collections[ns].chunks;
+      expect(chunks.length).to.equal(21);
+      expect(
+        !!chunks.find(
+          (tag) =>
+            tag ===
+            'too many chunks to print, use verbose if you want to force print'
+        )
+      ).to.equal(true);
```

Comment:
Totally optional, but I think you can simplify this to

```suggestion
      expect(chunks).to.include(
        'too many chunks to print, use verbose if you want to force print'
      );
```

or maybe use a stronger assertion like

```suggestion
      expect(
        chunks.indexOf(
          'too many chunks to print, use verbose if you want to force print'
        )
      ).to.equal(20);
```